experimental/gpu_monitoring/gpu_data_collector.py

```diff
@@ -140,6 +153,48 @@ def get_physicalhostname():
            return str(value[0])[2:][:-1]
 
 
+def get_counter_value(file_path):
+    file = open(file_path, "r")
+    value = file.read()
+    file.close()
+    return int(value)
+
+
+def infiniband_rate(current_counter, previous_counter, time_interval):
+    counter_delta = current_counter - previous_counter
+    if counter_delta < 0:
+        ib_counter_rate = 0
```

Comment:
I guess this is done because the counter resets when it reaches the bit-representable integer limit. 
Wouldn't it be better to set `ib_counter_rate = current_counter` instead of zero?
A more complete way would be to also add `limit - previous_counter`, but I do not know if we know the value of `limit`.

Reply:
Changed to
if counter_delta < 0:
        ib_counter_rate = int((2*64 + counter_delta) / time_interval))